homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,273 @@
+"""Support for Update."""
+from __future__ import annotations
+
+import asyncio
+import dataclasses
+import logging
+from typing import Any, Protocol
+
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import integration_platform, storage
+from homeassistant.helpers.typing import ConfigType
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "update"
+INFO_CALLBACK_TIMEOUT = 5
+STORAGE_VERSION = 1
+
+
+class UpdateFailed(HomeAssistantError):
```

Comment:
Although this is obviously the best name for this error, it's also already used for the update coordinator. This can cause integrations to accidentally auto-import the wrong one. So let's name it differently. Maybe `IntegrationUpdateFailed`

Reply:
Oh! :smile: